test/framework/src/main/java/org/elasticsearch/test/AbstractWireSerializingTestCase.java

```diff
@@ -48,35 +48,30 @@
     protected abstract Reader<T> instanceReader();
 
     /**
+     * Returns a {@link CopyFunction} that can be used to make an exact copy of
+     * the given instance. This defaults to a function that uses
+     * {@link #copyInstance(Writeable)} to create the copy.
+     */
+    protected CopyFunction<T> getCopyFunction() {
+        return (original) -> copyInstance(original);
+    }
+
+    /**
+     * Returns a {@link MutateFunction} that can be used to make create a copy
+     * of the given instance that is different to this instance. This defaults
+     * to null.
+     */
+    protected MutateFunction<T> getMutateFunction() {
```

Comment:
I agree that it would be better if this was mandatory but currently there are over 50 classes that extend this class and would need an implementation of this method added. I am happy to do this but do you think I should do it in this PR or in follow up PRs so there are a handful of smaller PRs rather than one large one?